simularium/simularium-website
# keyboard navigation: feature/focus states

Time estimate or Size
=======
_medium/small_

Problem
=======
Closes #575 and #576, advances #347 
Part of keyboard navigation and accessibility work.

Solution
========
Focus styling for action buttons has been reviewed with Lyndsay but is not reflected in master Figma yet.
Styling for primary/secondary buttons, and dropdown elements is there.
[Style guide](https://www.figma.com/design/u4lZYuVwbsFpzBikgwAVRb/Master-components-Simularium?node-id=1329-1588&node-type=instance&t=0ZsEwbvZECVFUkix-0)

Closing #576 because they already work, its clear how to tab through them now that the visual style is obivous.

Disabled buttons are taking focus/styling for now, this can be discussed more in future as we get into navigation behavior.

* New feature (non-breaking change which adds functionality)

### CustomDropdown
Wrapper for our Dropdowns, unifies the stylesheets, just a slight DRYing now, but will be very useful going forward.

Steps to Verify:
----------------
1. Tab through header and try action buttons (AICS home link, Simularium home link, Download, Share)
2. Open dropdowns and check on hover and focus styling (focus styling is difficult to maintain sometimes with mouse interactions. Easiest way to see it is the use the buttons that open the submenus.

Applying focus styling to ALL interactive elements is out of scope for this PR (e.g. submenu items in drop-downs)

Screenshots (optional):
-----------------------
<img width="171" alt="Screenshot 2024-10-16 at 2 51 56 PM" src="https://github.com/user-attachments/assets/14b4773c-af5f-4ca2-95d2-8f017138e151">
<img width="246" alt="Screenshot 2024-10-16 at 2 51 50 PM" src="https://github.com/user-attachments/assets/97e02c74-eaa9-4083-9039-c5cf0879e770">
<img width="235" alt="Screenshot 2024-10-16 at 2 51 43 PM" src="https://github.com/user-attachments/assets/f3f842ab-8e09-4125-8960-0053841119cf">
